src/sql/util.py

```diff
@@ -363,3 +365,123 @@ def show_deprecation_warning():
+    from current to root directory
+    """
+    current = Path().resolve()
+    while not (current / file_name).exists():
+        if current == current.parent:
+            return None
+
+        current = current.parent
+    display.message(f"Found {file_name} from '{current}'")
+
+    return str(Path(current, file_name))
+
+
+def find_close_match_config(word, possibilities, n=3):
+    """Finds closest matching configurations and displays message"""
+    closest_matches = difflib.get_close_matches(word, possibilities, n=n)
+    if not closest_matches:
+        display.message(
+            f"'{word}' is an invalid configuration. Please review our "
+            "configuration guideline: "
+            "https://jupysql.ploomber.io/en/latest/api/configuration.html#options"
+        )
+    else:
+        display.message(
+            f"'{word}' is an invalid configuration. Did you mean "
+            f"{pretty_print(closest_matches, last_delimiter='or')}?"
+        )
+
+
+def get_line_content_from_toml(file_path, line_number):
+    """
+    Locates a line that error occurs when loading a toml file
+    and returns the line, key, and value
+    """
+    with open(file_path, "r") as file:
+        lines = file.readlines()
+        eline = lines[line_number - 1].strip()
+        ekey, evalue = None, None
+        if "=" in eline:
+            ekey, evalue = map(str.strip, eline.split("="))
+        return eline, ekey, evalue
+
+
+def load_toml(file_path):
+    """
+    Returns toml file content in a dictionary format
+    and raises error if it fails to load the toml file
+    """
+    try:
+        with open(file_path, "r") as file:
+            content = file.read()
+            return toml.loads(content)
+    except toml.TomlDecodeError as e:
+        eline, ekey, evalue = get_line_content_from_toml(file_path, e.lineno)
+        if "Duplicate keys!" in str(e):
+            raise ValueError(f"Duplicate key found : '{ekey}'") from None
+        elif "Only all lowercase booleans" in str(e):
+            raise ValueError(
+                f"Invalid value '{evalue}' in '{eline}'. "
+                "Valid boolean values: true, false"
+            ) from None
+        elif "invalid literal for int()" in str(e):
+            raise ValueError(
+                f"Invalid value '{evalue}' in '{eline}'. "
+                "To use 'str' value, enclose it with ' or \"."
+            ) from None
+        else:
+            raise e from None
+
+
+def get_user_configs(file_path, section_names):
+    """
+    Returns saved configuration settings in a toml file from given file_path
+
+    Parameters
+    ----------
+    file_path : str
+        file path to a toml file
+    section_names : list
+        section names that contains the configuration settings
+
+    Returns
+    -------
+    dict
+        saved configuration settings
+    """
+    data = load_toml(file_path)
+    while section_names:
+        actual, given = section_names.pop(0), data.keys()
```

Comment:
Please rename `actual`, `given` to something more meaningful. I'm assuming `given` implies the one in `pyproject.toml`?
can u add an example of how `section_names` look like?

Reply:
- example add in the docstring of `get_user_configs`
- `actual` changed to `name_to_find`
- `given` changed to `names_from_user`